### DISCUSSION TITLE
Invalid file descriptor to ICU data received

### DISCUSSION THREAD

**Discussion by pangao1990** [pyinstaller/pyinstaller#7317] [Help]
我看到很多朋友和我一样遇到这个问题，而网上很难找到解决方案，这里我放出我的解决办法，供大家参考。我发现这个错误 ICU data 文件指的就是cefpython3缺失icudtl.dat。而这个文件在python3安装的cefpython3模块里就有。Pyinstaller在打包的时候，丢失了。得手动设置将这个文件打包进去，这个bug就解决了。

详情可以查看这里  https://github.com/pangao1990/vue-pywebview-pyinstaller/issues/5

**Comment by bwoodsend** [comment]
Good to know.

**Reply by baight** [reply]
我也遇到了同样的问题，解决方案如下：
1，找到cefpython3目录。通过pip show cefpython3可找到该目录，如：C:\Users\baight\AppData\Roaming\Python\Python38\site-packages\cefpython3。
2，复制 cefpython3目录下的 icudtl.dat、natives_blob.bin 到 .exe目录下的的 cefpython3 目录里。
3，复制 cefpython3目录下的 所有文件 到 .exe目录里。（经验证，真正需复制的文件是 locales、cef.pak、cef_100_percent.pak、cef_200_percent.pak、cef_extensions.pak、chrome_elf.dll、icudtl.dat、libcef.dll、libEGL.dll、libGLESv2.dll、natives_blob.bin、subprocess.exe、v8_context_snapshot.bin）。

**Reply by lktlktlkt** [reply]
第2步的cefpython3 到 .exe目录下的的 cefpython3是什么什么意思

**Reply by baight** [reply]
1，找到的python系统的cefpython3目录；2，自己打包的exe目录里，也会有一个cefpython3目录。
第2步的意思，就是把1里边cefpython3目录里的东西，复制到2里边的cefpython3目录。
